src/main/java/org/folio/rest/impl/DataExportImpl.java

```diff
@@ -83,53 +74,33 @@ public void getDataExportJobExecutions(String query, int offset, int limit, Stri
   }
 
   @Override
-  public void postDataExportFileDefinitions(FileDefinition entity, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
-    succeededFuture()
-      .compose(ar -> fileDefinitionService.save(entity.withStatus(Status.NEW), tenantId))
-      .map(PostDataExportFileDefinitionsResponse::respond201WithApplicationJson)
-      .map(Response.class::cast)
-      .otherwise(ExceptionToResponseMapper::map)
-      .setHandler(asyncResultHandler);
-  }
+  @Validate
+  public void getDataExportJobExecutionsDownloadByJobIdAndExportFileId(String jobId, String exportFileId,
+      Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
 
-  @Override
-  public void getDataExportFileDefinitionsByFileDefinitionId(String fileDefinitionId, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
-    succeededFuture()
-      .compose(ar -> fileDefinitionService.getById(fileDefinitionId, tenantId))
-      .map(optionalDefinition -> optionalDefinition.orElseThrow(() ->
-        new NotFoundException(String.format("File definition with id [%s] is not found", fileDefinitionId))))
-      .map(GetDataExportFileDefinitionsByFileDefinitionIdResponse::respond200WithApplicationJson)
+    succeededFuture().compose(ar -> fetchLink(jobId,exportFileId))
+      .map(GetDataExportJobExecutionsDownloadByJobIdAndExportFileIdResponse::respond200WithApplicationJson)
       .map(Response.class::cast)
       .otherwise(ExceptionToResponseMapper::map)
       .setHandler(asyncResultHandler);
   }
 
-  @Stream
-  @Override
-  public void postDataExportFileDefinitionsUploadByFileDefinitionId(String fileDefinitionId, InputStream entity, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+
+  private Future<FileDownload> fetchLink(String jobId, String exportFileId) {
+    ExportStorageService expService = exportStorageFactory.getExportStorageImplementation(System.getProperty("file.storage"));
```

Comment:
Please, retrieve `file.storage` on the ExportStorageFactory's side, not the caller side